public/app/features/explore/utils/panelsRegistry.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { RefObject } from 'react';
+
+import {
+  AbsoluteTimeRange,
+  DataFrame,
+  DataSourceApi,
+  GrafanaTheme2,
+  LoadingState,
+  SplitOpen,
+  TimeRange,
+  TimeZone,
+} from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { Collapse } from '@grafana/ui';
+// // TODO: probably needs to be exported from ui directly
+import { FilterItem, FILTER_FOR_OPERATOR, FILTER_OUT_OPERATOR } from '@grafana/ui/src/components/Table/types';
+
+import { ExploreGraphStyle, ExploreId } from '../../../types';
+import { ExploreGraph } from '../ExploreGraph';
+import { ExploreGraphLabel } from '../ExploreGraphLabel';
+import LogsContainer from '../LogsContainer';
+import { NodeGraphContainer } from '../NodeGraphContainer';
+import TableContainer from '../TableContainer';
+import { TraceViewContainer } from '../TraceView/TraceViewContainer';
+
+export function getPanelForVisType(visType: string): React.ComponentType<Props> | undefined {
+  // TODO this is not much dynamic at the moment but it's a start of creating a common interface
+  switch (visType) {
+    case 'graph': {
+      return GraphPanel;
+    }
+    case 'table': {
+      return TablePanel;
+    }
+
+    case 'nodeGraph': {
+      return NodeGraphPanel;
+    }
+
+    case 'logs': {
+      return LogsPanel;
+    }
+
+    case 'trace': {
+      return TraceViewPanel;
+    }
+    default: {
+      // Probably ok but maybe it makes sense to throw or show some info message
+      return TablePanel;
+    }
+  }
+}
+
+interface Props {
```

Comment:
@Elfo404 seems like this is going to be the interface :(. Seems awfully lots of things to pass to every panel and still some of them pull data directly from redux store. Will have to look at this later on to see if we can limit it with some other means (like theme probably does not need to be propagated like this) but let me know if you have some better out of the box idea.